myriad-scheduler/src/main/java/com/ebay/myriad/scheduler/event/handlers/ResourceOffersEventHandler.java

```diff
@@ -74,20 +76,35 @@ public void onEvent(ResourceOffersEvent event, long sequence,
     SchedulerDriver driver = event.getDriver();
     List<Offer> offers = event.getOffers();
 
+    // Sometimes, we see that mesos sends resource offers before Myriad receives
+    // a notification for "framework registration". This is a simple defensive code
+    // to not process any offers unless Myriad receives a "framework registered" notification.
+    if (schedulerState.getFrameworkID() == null) {
+      LOGGER.warn("Received {} offers, but declining them since Framework ID is not yet set", offers.size());
+      for (Offer offer : offers) {
+        driver.declineOffer(offer.getId());
+      }
+      return;
+    }
     LOGGER.info("Received offers {}", offers.size());
     LOGGER.debug("Pending tasks: {}", this.schedulerState.getPendingTaskIds());
     driverOperationLock.lock();
     try {
       for (Iterator<Offer> iterator = offers.iterator(); iterator.hasNext();) {
         Offer offer = iterator.next();
+        NodeTask nodeTask = schedulerState.getNodeTask(offer.getSlaveId());
```

Comment:
Yes. We need to set the slave attributes in NodeTask and use it later on (when a flexdown is requested with constraints).

I originally had this code at the place where we are launching the tasks. However, I heard Mesos is working on doing dynamic slave attributes. If the attributes on a slave change after Myriad launches the tasks, the new offers from the same slave may have a different set of attributes. I'm ok to move this down to the place where we perform launch tasks. @adam-mesos thoughts?